From: SeongJae Park <sj@kernel.org>
Date: Mon, 20 Jan 2025 13:43:21 -0800
Subject: [PATCH] ==== damon_call() for online parameter commit ====

mm/damon/sysfs: commit parameters online via damon_call()

Due to the lack of ways to synchronously access DAMON internal data,
DAMON sysfs interface is using damon_callback hooks with its own
synchronization mechanism.  The mechanism is built on top of
damon_callback hooks in an ineifficient and complicated way.

Patch series "mm/damon: replace most damon_callback usages in sysfs with
new core functions", which starts with commit e035320fd38e
("mm/damon/sysfs-schemes: remove unnecessary schemes existence check in
damon_sysfs_schemes_clear_regions()") introduced two new DAMON kernel
API functions that providing the synchronous access, replaced most
damon_callback hooks usage in DAMON sysfs interface, and cleaned up
unnecessary code.

Continue the replacement and cleanup works.  Update the last DAMON
sysfs's usage of its own synchronization mechanism, namely online DAMON
parameters commit, to use damon_call() instead of the damon_callback
hooks and the hard-to-maintain own synchronization mechanism.  Then
remove the no more be used code due to the change, and more unused code
that just not yet cleaned up.

The first four patches of this series do the replacement.  Then,
following three patches (patches 5-7) remove the DAMON sysfs interface'
own synchronization mechanism and its usages, which is no more be used
by anyone due to the first four patches.  Finally, six patches (8-13)
doing more cleanup of no more valid comment and no more used code
follow.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_cw2l52w8 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_cw2l52w8

diff --git a/hkml_cv_bogus/hkml_cv_bogus_cw2l52w8 b/hkml_cv_bogus/hkml_cv_bogus_cw2l52w8
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

